modules/javafx.graphics/src/main/java/com/sun/glass/ui/headless/HeadlessRobot.java

```diff
@@ -0,0 +1,406 @@
+                .filter(win -> win.getView() != null)
+                .filter(win -> !win.isClosed())
+                .filter(win -> !win.isMinimized()).toList();
+        if (windows.isEmpty()) return null;
+        if (windows.size() == 1) return (HeadlessWindow)windows.get(0);
+        return (HeadlessWindow)windows.get(windows.size() -1);
+    }
+
+    private HeadlessWindow getFocusedWindow() {
+        List<Window> windows = Window.getWindows().stream()
+                .filter(win -> win.getView()!= null)
+                .filter(win -> !win.isClosed())
+                .filter(win -> win.isFocused()).toList();
+        if (windows.isEmpty()) return null;
+        if (windows.size() == 1) return (HeadlessWindow)windows.get(0);
+        return (HeadlessWindow)windows.get(windows.size() -1);
+    }
+
+    private HeadlessWindow getTargetWindow(double x, double y) {
+        List<Window> windows = Window.getWindows().stream()
+                .filter(win -> win.getView()!= null)
+                .filter(win -> !win.isClosed())
+                .filter(win -> (x >= win.getX() && x <= win.getX() + win.getWidth()
+                        && y >= win.getY() && y <= win.getY()+ win.getHeight())).toList();
+        if (windows.isEmpty()) {
+            return null;
+        }
+        if (windows.size() == 1) return (HeadlessWindow)windows.get(0);
+        return (HeadlessWindow)windows.get(windows.size() -1);
+    }
+
+    int getModifiers(MouseButton... buttons) {
+        int modifiers = KeyEvent.MODIFIER_NONE;
+        for (int i = 0; i < buttons.length; i++) {
+            switch (buttons[i]) {
+                case PRIMARY:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_PRIMARY;
+                    break;
+                case MIDDLE:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_MIDDLE;
+                    break;
+                case SECONDARY:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_SECONDARY;
+                    break;
+                case BACK:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_BACK;
+                    break;
+                case FORWARD:
+                    modifiers |= KeyEvent.MODIFIER_BUTTON_FORWARD;
+                    break;
+            }
+        }
+        return modifiers;
+    }
+
+    int getGlassEventButton(MouseButton[] buttons) {
+        if ((buttons == null) || (buttons.length == 0)) return 0;
+        return getGlassEventButton(buttons[0]);
+    }
+
+    int getGlassEventButton(MouseButton button) {
+        if (button == MouseButton.SECONDARY) return MouseEvent.BUTTON_RIGHT;
+        if (button == MouseButton.PRIMARY) return MouseEvent.BUTTON_LEFT;
+        if (button == MouseButton.MIDDLE) return MouseEvent.BUTTON_OTHER;
+        if (button == MouseButton.BACK) return MouseEvent.BUTTON_BACK;
+        if (button == MouseButton.FORWARD) return MouseEvent.BUTTON_FORWARD;
+        return MouseEvent.BUTTON_NONE;
```

Comment:
Use an expression to get compile-time exhaustiveness checks:
```suggestion
        return switch (button) {
            case NONE -> MouseEvent.BUTTON_NONE;
            case PRIMARY -> MouseEvent.BUTTON_LEFT;
            case MIDDLE -> MouseEvent.BUTTON_OTHER;
            case SECONDARY -> MouseEvent.BUTTON_RIGHT;
            case BACK -> MouseEvent.BUTTON_BACK;
            case FORWARD -> MouseEvent.BUTTON_FORWARD;
        };
```